components/table/src/table/table.component.ts

```diff
@@ -341,7 +341,7 @@ export class NzTableComponent<T> implements OnInit, OnDestroy, OnChanges, AfterV
       .observe(this.elementRef)
       .pipe(
         map(([entry]) => {
-          const { width } = entry.target.getBoundingClientRect();
+          const { width } = entry.contentRect;
```

Comment:
I think the problem is that nzResizeObserver is triggered at wrong time(entry.target equals `nz-table`), so we can try to remove `style.width.px` in `table-fixed-row.component.ts` when it is used for empty or try to figure out why `NzResizeObserver` returns the unrendered elements.